where_my_money/transactions.py

```diff
@@ -0,0 +1,245 @@
+    def from_dict(dct: dict) -> "BaseTransaction":
+        """Base from dict method"""
+        raise NotImplementedError
+
+
+class GenericTransaction(BaseModel):
+    """Mapping for all transactions"""
+
+    ### dimensions
+    source: str
+    account_type: str
+    payment_type: str
+    payment_instruction: str | None
+    payment_category: str | None
+    ### amount
+    currency: str
+    amount: float
+    fee: float | None
+    balance: float | None
+    ### descriptions
+    description: str | None
+    message_for_sender: str | None
+    message_for_recipient: str | None
+    ### counterparty
+    counterparty: str | None
+    counterparty_account_number: str | None
+    counterparty_account_name: str | None
+    business: str | None
+    ### dates
+    payment_date: datetime | None
+    started_date: datetime | None
+    due_date: datetime | None
+    completed_date: datetime | None
+    ### bool
+    is_completed: bool | None  # state + is_completed
+
+
+class RevolutTransaction(BaseModel, BaseTransaction):
+    """Mapping for Revolut transactions"""
+
+    class Config:
+        """Config for Revolut transactions"""
+
+        allow_population_by_alias = True
+        anystr_strip_whitespace = True
+
+    __file_encoding__: str = "utf-8"
+    __delimiter__: str = ","
+    source: str = "revolut"
+    account_type: str = "basic"
+    payment_type: str = Field(alias="Type")
+    started_date: datetime = Field(alias="Started Date")
+    completed_date: datetime = Field(alias="Completed Date")
+    description: str | None = Field(alias="Description")
+    amount: float = Field(alias="Amount")
+    fee: float = Field(alias="Fee")
+    currency: str = Field(alias="Currency")
+    state: str = Field(alias="State")
+    balance: float | None = Field(alias="Balance")
+
+    def to_generic_transaction(self) -> "GenericTransaction":
+        """Mapping to generic transaction"""
+
+        return GenericTransaction(
+            source=self.source,
+            account_type=self.account_type,
+            payment_type=self.payment_type,
+            payment_instruction=None,
+            payment_category=None,
+            currency=self.currency,
+            amount=self.amount,
+            fee=self.fee,
+            balance=self.balance,
+            description=self.description,
+            message_for_sender=None,
+            message_for_recipient=None,
+            counterparty=None,
+            counterparty_account_name=None,
+            counterparty_account_number=None,
+            business=None,
+            payment_date=None,
+            started_date=self.started_date,
+            due_date=None,
+            completed_date=self.completed_date,
+            is_completed=self.state == "COMPLETED",
+        )
+
+    @staticmethod
+    def from_dict(dct: dict) -> "RevolutTransaction":
+        """Mapping from dict"""
+
+        for key, value in dct.items():
+            if value == "":
+                dct[key] = None
```

Comment:
Here, on the contrary, I'd refactor this into a function, because you actually call this more often.

Reply:
okay, done